app/helpers/my_modules_helper.rb

```diff
@@ -112,4 +112,63 @@ def my_module_due_status(my_module, datetime = DateTime.current)
       ''
     end
   end
+
+  def serialize_assigned_my_module_value(my_module)
+    [
+      serialize_assigned_my_module_team_data(my_module.team),
+      serialize_assigned_my_module_project_data(my_module.project),
+      serialize_assigned_my_module_experiment_data(my_module.experiment),
+      serialize_assigned_my_module_data(my_module)
+    ]
+  end
+
+  private
+
+  def serialize_assigned_my_module_team_data(team)
+    {
+      type: team.class.name.underscore,
+      value: team.name,
+      url: projects_path(team: team.id),
+      archived: false
+    }
+  end
+
+  def serialize_assigned_my_module_project_data(project)
+    {
+      type: project.class.name.underscore,
+      value: project.name,
+      url: project_path(project, view_mode: view_mode(project.archived?)),
+      archived: project.archived?
+    }
+  end
+
+  def serialize_assigned_my_module_experiment_data(experiment)
+    {
+      type: experiment.class.name.underscore,
+      value: experiment.name,
+      url: experiment_url(experiment),
+      archived: experiment.archived_branch?
+    }
+  end
+
+  def serialize_assigned_my_module_data(my_module)
+    {
+      type: my_module.class.name.underscore,
+      value: my_module.name,
+      url: protocols_my_module_path(my_module, view_mode: view_mode(my_module.archived_branch?)),
+      archived: my_module.archived_branch?
+    }
+  end
+
+  def view_mode(archived)
```

Comment:
probably it's not worth extracting it into a separate method as it is used only twice